accounts-db/src/accounts_index.rs

```diff
@@ -1362,18 +1362,17 @@ impl<T: IndexValue, U: DiskIndexValue + From<T> + Into<T>> AccountsIndex<T, U> {
         self.account_maps.len()
     }
 
-    /// remove the earlier instances of each pubkey when the pubkey exists later in the `Vec`.
-    /// Could also be done with HashSet.
-    /// Returns `HashSet` of duplicate pubkeys.
-    fn remove_older_duplicate_pubkeys(
-        items: &mut Vec<(Pubkey, (Slot, T))>,
+    /// remove the earlier instances of each pubkey when consecutive positions of the same
+    /// pubkey exist in the `Vec`.
+    /// The input can be partially sorted, only adjacent duplicates are removed.
+    /// Returns `Vec` of duplicate pubkeys.
+    fn remove_adjacent_older_duplicate_pubkeys(
```

Comment:
I think we don't have any dups in the storage nowadays. 
Probably, we could skip this fn too.

Reply:
Interesting, indeed an assertion that duplicates are empty is passing on my validator... I will remove it and cleanup a couple more things around this code in separate PR.

For now, I just had to resolve conflict.